app/src/main/java/com/github/geohunt/app/ui/components/navigation/NavigationController.kt

```diff
@@ -199,5 +200,31 @@ fun NavigationController(
             )
             PrivacySettingsPage(onBack = { navController.popBackStack() }, viewModel)
         }
+
+        // Bounties
+        // Open a claim for a given bounty's challenge
+        composable(
+            "bounty-claim-challenge/{bountyId}/{challengeId}",
+            arguments = listOf(
+                navArgument("bountyId") { type = NavType.StringType },
+                navArgument("challengeId") { type = NavType.StringType })
+        ) { backStackEntry ->
+            val bid = backStackEntry.arguments?.getString("bountyId")!!
+            val cid = backStackEntry.arguments?.getString("challengeId")!!
+
+            BountyClaimChallenge(
+                bid = bid,
+                cid = cid,
+                onFailure = {
+                    Toast.makeText(context, "Something went wrong, failed to create challenge", Toast.LENGTH_LONG).show()
+                    Log.e("GeoHunt", "Fail to create challenge: $it")
+                    navController.popBackStack()
+                },
+                onClaimSubmitted = {
+                    navController.popBackStack()
+                    navController.navigate("challenge-view/$cid")
```

Comment:
Use the hidden route?

Reply:
As above, I have changed my code to use the hidden route.